### PR TITLE
Minor services naming fix

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.5.x
| Description?  | When looking at `HtaccessFileChecker` and `RobotsTextFileChecker` they seem to be checking if file is writable, but interface is a bit misleading. So it updates the naming.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | No behavior has changed. Page "Configure -> Shop parameters -> Traffic & Seo -> Seo & Urls" should work as before.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10845)

### CODE REVIEW

**Comment by sarjon** [comment]
@matks wdyt?

**Comment by PierreRambaud** [comment]
You can't change file names and methods names without introduce BC break. Do you know when these classes were added?

**Comment by sarjon** [comment]
> without introduce BC break

yes, they should be part of 1.7.5, thats why i want to change them before release. :)

**Comment by PierreRambaud** [comment]
In `isWritable` methods.
```php
return is_writable(dirname($filePath));
```
can be:
```php
return is_writable($this->rootDirectory);
```

Prevent useless usage of `dirname()`.
Otherwise I approved!

**Comment by sarjon** [comment]
> Prevent useless usage of dirname().

totally right, i intended to do this, but missed somehow. :(

**Comment by PierreRambaud** [comment]
And I just realize these classes have same methods, could be useless to have an abstract class? Wdyt?

**Review by PierreRambaud — DISMISSED** [review]
@matks wdyt?

**Comment by matks** [comment]
@PierreRambaud Well, except for one string these classes are identical.
So why even an abstract class ? We make create a class `FileChecker` and we can
- construct 2 services by proving the file name as constructor input
or
- construct 1 service and provide the filepath as argument in `isWritable($filepath)`

**Comment by sarjon** [comment]
i've implemented single specialized service instead of 2 generic now.

ping @PierreRambaud @matks for review. :)

File: src/Core/Util/Url/UrlFileChecker.php [hunk 1]
```diff
@@ -27,37 +27,52 @@
...
     {
-        $filePath = $this->rootDirectory . '/.htaccess';
+        $filePath = $this->fileDir . DIRECTORY_SEPARATOR . $fileName;
 
         if (file_exists($filePath)) {
             return is_writable($filePath);
         }
 
-        return is_writable(dirname($filePath));
+        return is_writable($this->fileDir);
```
Comment (PierreRambaud) [inline src/Core/Util/Url/UrlFileChecker.php:76]: Something I don't understand with this function:
If the file doesn't exists, why checking for writable directory?
Having a test on .htaccess should test the .htaccess, not the parent directory :/

Comment (sarjon) [reply src/Core/Util/Url/UrlFileChecker.php:76]: if file does not exist, but its directory is writable, we can say that file is writable (we dont care whether it exists or not, we know that it's writable).

but this is my guess. :)

**Review by matks — APPROVED** [review]

File: src/Core/Util/Url/UrlFileChecker.php (hunk 1 above)
Comment (PierreRambaud) [reply src/Core/Util/Url/UrlFileChecker.php:76]: I understand what you mean but watchout with linux permissions (sticky bit and acls), the file doesn't exists, so it's logically not writable :)

Comment (sarjon) [reply src/Core/Util/Url/UrlFileChecker.php:76]: well, if it was working in legacy and nobody raised an issue, i guess its okay? :)

Comment (PierreRambaud) [reply src/Core/Util/Url/UrlFileChecker.php:76]: PrestaShop must be compatible with complex architecture, this is maybe something we will have to change in the future :)

**Comment by matks** [comment]
Thank you @sarjon

**Pull request merged**